include/sys/vm_object.h

```diff
@@ -14,12 +16,13 @@
  */
 
 typedef struct vm_object {
-  mtx_t mtx;
+  rwlock_t mtx;
```

Comment:
I thought that in case when we are using the cow, reading access will be more common than the writing access so predictably rwlocks would be faster than mutexes.

Reply:
It is true that with cow reads from `vm_objects` will become more frequent, but in order to benefit performance, read accesses must be _far_ more frequent than write accesses, and read accesses must often occur concurrently. As for the first requirement, that may be the case, although I haven't looked through the code. As for the second, I don't think concurrent read accesses are likely, since we don't support SMP, so concurrent access can occur only when a thread is preempted while holding the lock.
I'm also a little concerned about the fact that `rwlock` uses `sleepq` functions, so there may be starvation and priority inversion issues, since you sometimes acquire a write lock while holding a mutex, e.g. in `vm_page_fault()`.

Reply:
Okay, I see what you mean. I agree with you. That's right. I also am worried a little about the fact that we don't use rwlocks in mimiker frequently, so they aren't tested well, so we don't know if they work properly for sure. I will try to split this PR to several smaller ones, so rwlocks would be separate change that maybe will be skipped.